src/lib/button/button.ts

```diff
@@ -57,17 +62,28 @@ export class MdIconButtonCssMatStyler {}
   selector: 'button[md-fab], button[mat-fab], a[md-fab], a[mat-fab]',
   host: {'class': 'mat-fab'}
 })
-export class MdFabCssMatStyler {}
+export class MdFab {
+  constructor(@Self() @Optional() button: MdButton, @Self() @Optional() anchor: MdAnchor) {
```

Comment:
Here and in MdMiniFab, looks like we want to use `forwardRef` since those components are defined further down the file. This caused problems internally, surprised our tests didn't have trouble with them.